pkgs/test/default.nix

```diff
@@ -31,6 +31,7 @@ with pkgs;
 
   fetchurl = callPackages ../build-support/fetchurl/tests.nix { };
   fetchpatch = callPackages ../build-support/fetchpatch/tests.nix { };
+  fetchpatch2 = callPackages ../build-support/fetchpatch/tests.nix { fetchpatch = fetchpatch2; };
```

Comment:
I think we should move this right into the test then we could also reduce the changes in all-packages.

Reply:
I think the current solution is decent. If we need something else, we can change it.